layers/+filetree/treemacs/packages.el

```diff
@@ -41,6 +41,10 @@
             treemacs-never-persist nil
             treemacs-goto-tag-strategy 'refetch-index
             treemacs-collapse-dirs treemacs-use-collapsed-directories)
+      (add-hook 'treemacs-mode-hook
+                (lambda ()
+                  (when treemacs--width-is-locked
+                    (treemacs-toggle-fixed-width))))
```

Comment:
Ah, sorry. I just tried to use `treemacs` and it turns out that I misunderstood the meaning of non-fixed width. It doesn't change the width automatically based on content, but rather allows the user to manually change the width of `treemacs` buffer.

Then my previous comment is not that critical, but I still think that it is a good idea to have an ability to enable fixed-width with a variable in `.spacemacs`. Plus, we already have several similar variables (e.g. `treemacs-use-follow-mode`).

Reply:
Your second comment is accurate. Makes sense, will do. I will make it a layer variable since it is part of the treemacs layer.

It's just that in the first place I'm having a lot of troubles imagining for what reason anybody would want it locked. But at the same time, it being implemented that way means that at least one person in the world has a reason for that so a configuration variable makes sense.